lib/segment/types.ex

```diff
@@ -0,0 +1,138 @@
+defmodule Segment.Analytics.Types do
+  def common_fields do
+    [
+      :anonymousId,
+      :context,
+      :integrations,
+      :timestamp,
+      :userId,
+      :version
+    ]
+  end
+end
+
+defmodule Segment.Analytics.Track do
+  @method "track"
+
+  defstruct Segment.Analytics.Types.common_fields() ++
+              [
+                :event,
+                :properties,
+                type: @method
+              ]
+
+  def new(attrs) do
+    struct(__MODULE__, attrs)
+  end
+end
+
+defmodule Segment.Analytics.Identify do
+  @method "identify"
+
+  defstruct Segment.Analytics.Types.common_fields() ++
+              [
+                :traits,
+                type: @method
+              ]
+
+  def new(attrs) do
+    struct(__MODULE__, attrs)
+  end
+end
+
+defmodule Segment.Analytics.Alias do
+  @method "alias"
+
+  defstruct Segment.Analytics.Types.common_fields() ++
+              [
+                :previousId,
+                type: @method
+              ]
+
+  def new(attrs) do
+    struct(__MODULE__, attrs)
+  end
+end
+
+defmodule Segment.Analytics.Page do
+  @method "page"
+
+  defstruct Segment.Analytics.Types.common_fields() ++
+              [
+                :name,
+                :properties,
+                type: @method
+              ]
+
+  def new(attrs) do
+    struct(__MODULE__, attrs)
+  end
+end
+
+defmodule Segment.Analytics.Screen do
+  @method "screen"
+
+  defstruct Segment.Analytics.Types.common_fields() ++
+              [
+                :name,
+                :properties,
+                type: @method
+              ]
+
+  def new(attrs) do
+    struct(__MODULE__, attrs)
+  end
+end
+
+defmodule Segment.Analytics.Group do
+  @method "group"
+
+  defstruct Segment.Analytics.Types.common_fields() ++
+              [
+                :groupId,
+                :traits,
+                type: @method
+              ]
+
+  def new(attrs) do
+    struct(__MODULE__, attrs)
+  end
+end
+
+defmodule Segment.Analytics.Context do
+  @library_name Mix.Project.get().project[:description]
```

Comment:
I can't remember if `Mix.Project` is always available at compile time in prod? If I recall correctly at Frame these not got set but I could be wrong

Reply:
good point. this was from the old library. I think it might not be though

Reply:
i created a release with `MIX_ENV=prod mix release` and ran it through iex and they were being set. any idea how to create a test that might make this fail?

Reply:
Probably being set at compile time then. I'd say leave in.